// @markslides/markdown-it-mermaid/dist/index.mjs
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};

// src/index.ts
import mermaid from "mermaid";
var mermaidChart = (code) => {
  try {
    mermaid.parse(code, { suppressErrors: true });
    return `<div class="mermaid">${code}</div>`;
  } catch ({ str, hash }) {
    return `<pre>${str}</pre>`;
  }
};
var isDarkMode = false;
var markdownItMermaid = (md, config) => {
  mermaid.initialize(__spreadValues({
    theme: isDarkMode ? "default" : "dark",
    darkMode: isDarkMode,
    fontFamily: "monospace",
    // fontFamily: 'ui-monospace',
    // altFontFamily: 'monospace',
    startOnLoad: true
  }, config));
  mermaid.registerIconPacks([
    {
      name: "logos",
      loader: () => fetch(
        "https://unpkg.com/@iconify-json/logos@1/icons.json"
      ).then((res) => res.json())
    }
  ]);
  md.mermaid = mermaid;
  const original = md.renderer.rules.fence || function(tokens, idx, options, env, self) {
    return self.renderToken(tokens, idx, options);
  };
  md.renderer.rules.fence = (tokens, idx, options, env, self) => {
    const token = tokens[idx];
    const code = token.content.trim();
    if (token.info === "mermaid") {
      return mermaidChart(code);
    }
    return original(tokens, idx, options, env, self);
  };
};
var src_default = markdownItMermaid;
export {
  src_default as default
};
